src/main/java/org/opensearch/security/dlic/rest/validation/PasswordValidator.java

```diff
@@ -0,0 +1,181 @@
+                );
+            }
+            return ValidationResult.INVALID;
+        }
+        if (password.length() > MAX_LENGTH) {
+            if (logger.isDebugEnabled()) {
+                logger.debug(
+                        "Password is too long, the maximum required length is {}, but current length is {}",
+                        MAX_LENGTH,
+                        password.length()
+                );
+            }
+            return ValidationResult.INVALID;
+        }
+        if (Objects.nonNull(passowrdRegexpPattern)
+                && !passowrdRegexpPattern.matcher(password).matches()) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Regex does not match password");
+            }
+            return ValidationResult.INVALID;
+        }
+        final Strength strength = zxcvbn.measure(password, ImmutableList.of(username));
+        if (strength.getScore() < scoreStrength.score()) {
+            logger.debug(
+                    "Password is weak the required score is {}, but current is {}",
+                    scoreStrength,
+                    ScoreStrength.fromScore(strength.getScore())
+            );
+            return ValidationResult.WEAK;
+        }
+        final boolean similar = strength.getSequence()
+                .stream()
+                .anyMatch(USERNAME_SIMILARITY_CHECK);
+        if (similar) {
+            logger.debug("Password is too similar to the user name {}", username);
+            return ValidationResult.SIMILAR;
+        }
+        return ValidationResult.VALID;
+    }
+
+    public enum ValidationResult {
+        VALID,
+
+        INVALID,
+
+        SIMILAR,
+
+        WEAK;
+    }
+
+    public enum ScoreStrength {
+
+        WEAK(0, "too guessable: risky password"),
+        FAIR(1, "very guessable: protection from throttled online attacks"),
+        GOOD(2, "somewhat guessable: protection from unthrottled online attacks"),
+        STRONG(3, "safely unguessable: moderate protection from offline slow-hash scenario"),
+        VERY_STRONG(4, "very unguessable: strong protection from offline slow-hash scenario");
+
+        private final int score;
+
+        private final String description;
+
+        private ScoreStrength(final int score, final String description) {
+            this.score = score;
+            this.description = description;
+        }
+
+        public static ScoreStrength fromScore(final int score) {
+            if (WEAK.score == score) {
+                return WEAK;
+            } else if (FAIR.score == score) {
+                return FAIR;
+            } else if (GOOD.score == score) {
+                return GOOD;
+            } else if (STRONG.score == score) {
+                return STRONG;
+            } else if (VERY_STRONG.score == score) {
+                return VERY_STRONG;
+            } else {
+                throw new IllegalArgumentException("Unknown score " + score);
+            }
+        }
```

Comment:
Enum's `.values()` makes this much cleaner
```suggestion
        public static ScoreStrength fromScore(final int score) {
            for (ScoreStrength strength : ScoreStrength.values()) {
                if (strength.score == score) {
                    return strength;
                }
            }
            throw new IllegalArgumentException("Unknown score " + score);
        }
```

Reply:
sure